analyzers/src/SonarAnalyzer.CSharp/Rules/LoopsAndLinq.cs

```diff
@@ -67,12 +67,21 @@ node switch
                 _ => null
             };
 
-        private static bool CanIfStatementBeMoved(IfStatementSyntax ifStatementSyntax) =>
-            ifStatementSyntax.Else == null
-            && ifStatementSyntax.Condition is InvocationExpressionSyntax invocationExpressionSyntax
-            && !invocationExpressionSyntax.DescendantNodes()
-                                          .OfType<ArgumentSyntax>()
-                                          .Any(argument => argument.RefOrOutKeyword.IsAnyKind(SyntaxKind.OutKeyword, SyntaxKind.RefKeyword));
+        private static bool CanIfStatementBeMoved(IfStatementSyntax ifStatementSyntax)
+        {
+            return ifStatementSyntax.Else == null && (ConditionValidIsPattern() || ConditionValidInvocation());
+
+            bool ConditionValidIsPattern() => IsPatternExpressionSyntaxWrapper.IsInstance(ifStatementSyntax.Condition)
+                                         && !ifStatementSyntax.Condition.DescendantNodes()
+                                                                        .Any(d => d.IsAnyKind(SyntaxKindEx.VarPattern,
```

Comment:
Since someone might use a `var` or a local name on an `is` pattern, i removed these cases from raising issues, since the value will be useful only inside the `Where` clause, therefore useless. 

See the new UTs for more explanation.

Reply:
Looks very good, I would only ask you to add more UTs with other patterns that use this VarPattern or the Designations, to check that it works for other patterns and not just the ListPattern.